sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/InMemoryColumnarQuerySuite.scala

```diff
@@ -503,7 +503,7 @@ class InMemoryColumnarQuerySuite extends QueryTest with SharedSQLContext {
             case plan: InMemoryRelation => plan
           }.head
           // InMemoryRelation's stats is file size before the underlying RDD is materialized
-          assert(inMemoryRelation.computeStats().sizeInBytes === 740)
+          assert(inMemoryRelation.computeStats().sizeInBytes === 800)
```

Comment:
Our optimizer uses the statistics to decide the plans (e.g., in join algorithm selection). Thus, the plans could be completely different if the file size increases by 8 percents. Could you give us more contexts? cc @rdblue

Reply:
Parquet fixed a problem with value ordering in statistics, which required adding new metadata min and max fields. For older readers, Parquet also writes the old values when it makes sense to. This is a slight increase in overhead, which is more noticeable when files contain just a few records.

Don't be alarmed at the percentage difference here, it is just a small file. Parquet isn't increasing file sizes by 8%, that would be silly.

Reply:
Then it looks fine, but the new metadata max/min fields are added in file metadata, column metadata and page header metadata? Any formula we can use to calculate the size increase?

Reply:
This is data dependent so it is hard to estimate. We write the stats for older readers when the type uses a signed sort order, so it is limited to mostly primitive types and won't be written for byte arrays or utf8 data. That limits the size to 16 bytes + thrift overhead per page and you might have about 100 pages per row group. So 1.5k per 128MB, which is about 0.001%.